website_event_membership_restriction/views/event_templates.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2024 Tecnativa - Pilar Vargas
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <template
+        id="registration_template"
+        inherit_id="website_event.registration_template"
+        priority="999"
+    >
+        <xpath
+            expr="//form//div[@id='o_wevent_tickets_collapse']//button"
+            position="attributes"
+        >
+            <attribute name="class">d-none</attribute>
+        </xpath>
+        <xpath
+            expr="//form//div[@id='o_wevent_tickets_collapse']//button"
+            position="after"
+        >
+            <button
+                type="submit"
+                t-att-class="'btn btn-primary o_wait_lazy_js a-submit ' + (event.registration_membership_only and (not request.env.user or request.env.user.partner_id.commercial_partner_id.membership_state not in ['paid', 'free'] or request.env.user.partner_id.membership_state not in ['paid', 'free']) and 'disabled' or '')"
```

Comment:
Do this condition make sense? Should it be something like
```
event.registration_membership_only and not request.env.user._is_public() and (... or ...)

Reply:
Change made to check with ‘is_public()’.

This way logging will be disabled for the following cases:

- registration_membership_only = true and user not logged in.
- registration_membership_only = true and partner state not free, paid
- registration_membership_only = true and state of partner commercial not free, paid


thanks!